src/DurableTask.Netherite/PartitionState/PrefetchState.cs

```diff
@@ -74,13 +74,19 @@ public override void Process(WaitRequestReceived waitRequestEvent, EffectTracker
         void ProcessClientRequestEventWithPrefetch(ClientRequestEventWithPrefetch clientRequestEvent, EffectTracker effects)
         {
             if (clientRequestEvent.Phase == ClientRequestEventWithPrefetch.ProcessingPhase.Read)
-            {           
-                this.Partition.Assert(!this.PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString), "PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString)");
-
-                // Issue a read request that fetches the instance state.
-                // We have to buffer this request in the pending list so we can recover it.
+            {
+                if (!this.PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString))
+                        {
+                    // Issue a read request that fetches the instance state.
+                    // We buffer this request in the pending list so we can recover it, and can filter duplicates
+                    // (as long as the duplicates appear soon after the original)
 
-                this.PendingPrefetches.Add(clientRequestEvent.EventIdString, clientRequestEvent);
+                    this.PendingPrefetches.Add(clientRequestEvent.EventIdString, clientRequestEvent);
+                }
+                else
+                {
+                    return; // this is a duplicate. Ignore it.
+                }
```

Comment:
nit: given that this is rare - should we log it? Might be helpful to know if EH is acting up and duplicating events a lot.